src/CovidAppointmentTable.js

```diff
@@ -90,13 +141,24 @@ export default function CovidAppointmentTable() {
 				}
 				label="Only show locations with available appointments"
 			/>
+			<div >
+				<TextField id="outlined-basic" label="Enter Zipcode" variant="outlined" onChange={e => setZipcode(e.target.value)} value={zipcode}/>
+				<Button className={classes.filterButtons} variant="contained" color="secondary" onClick={getZipcodeLocation}>Search By Zip</Button>
+				<Button className={classes.filterButtons} variant="contained" color="secondary" startIcon={<GpsFixedIcon/>} onClick={getCurrentLocation}>Use My Location</Button>
+			</div>
+			{!loading && data.length === 0 ?
+				<div className={classes.notFoundBox}>
+					<h2 className={classes.locationsNotFound}>No Locations Found</h2>
+					<Button className={classes.filterButtons} variant="contained" color="secondary" onClick={resetData}>Show All Locations</Button>
+				</div> :(
+			<Loader loaded={!!data && data.length > 0 && !loading}>
 			{formattedData.map((entry) => {
 				return (
 					<div className={classes.cardBox}>
 						<Card>
 							<CardHeader
 								title={<div>{entry.location}</div>}
-								subheader={<div>{entry.city}</div>}
+								subheader={<><div>{entry.city}</div>{entry.distance && <div><RoomOutlined/>{Number.parseFloat(entry.distance).toFixed(1)} miles away</div>}</>}
```

Comment:
I made a comment above, but I don't think we should show this unless a ZIP is set.

Reply:
What if they use the `Use My Location`? That would provide the coordinates without the zipcode input being provided.